### ISSUE TITLE
Add JWT token connection option 

### ISSUE THREAD

**Issue by LastTribunal** [Azure/elastic-db-tools#161]
This option exists in Azure SQL, and should unconditionally be part of this library

**Comment by tdesmet** [comment]
I was looking for a way to pass the token obtained from Azure AD for authentication instead of username and password. Is there any plan or roadmap to support this?

I was planning on using this together with Managed Service Identity and the Microsoft.Azure.Services.AppAuthentication library...

**Comment by jaredmoo** [comment]
Fixing this in #189

**Comment by neilwebb** [comment]
Hi,
I can see some changes to support shard management and retrieval by key in #189 using an AccessToken via the factory delegate in the newly introduced SqlConnectionInfo but are there there plans as part of this to update MultiShardConnection too so that the underlying shard connections can also use AD auth?

**Comment by jaredmoo** [comment]
Yes, MultiShardConnection support is also needed and will be done.

**Comment by LastTribunal** [comment]
@jaredmoo 
I am still using your original sharding library and with Token Based Auth
All i had to do was upgrade your code to .net 4.5 and create a static variable which I expose to the caller and which I also added to your SQLConnection calls in you code (only two places).  The caller manages the token using native AD token libraries.

**Comment by mholo65** [comment]
MultiShardConnection fixed by #201 

@jaredmoo any chance to get this merged and get hands on a pre-release soonish (alternatively release)? We are removing SQL Credentials from an App and the Elastic Scale is the only thing currently blocking us. Or are there currently any workarounds for passing AccessToken in version 2.3.0?

**Comment by jaredmoo** [comment]
I merged it :) sorry for the delay with the release, I have been running into challenges with our signing process that are taking a long time to work through, that's the only thing blocking release. I suggest you could build from master branch yourself for now as a workaround.

**Comment by hayesst** [comment]
Any updates on when this will be released?

**Comment by hayesst** [comment]
@jaredmoo  - Are there any plans to release these updates?  We'd like to implement and would prefer to do it via an official release path.  Any info would be helpful.  Thanks

**Comment by stuartpa** [comment]
2.4.0 released to nuget.org:

https://www.nuget.org/packages/Microsoft.Azure.SqlDatabase.ElasticScale.Client/2.4.0

Release notes:

https://github.com/Azure/elastic-db-tools/releases/tag/v2.4.0

Let me know if you hit any issues
